source/slang/slang-serialize-ir.cpp

```diff
@@ -14,66 +17,299 @@
 //
 #include "slang-serialize-ir.cpp.fiddle"
 
+// If USE_RIFF is set, then we serialize using the RIFF backend, it's the
+// slowest option
+#define USE_RIFF 0
+// If we are serializing using Fossil, DIRECT_FROM_FOSSIL will make it so that
+// we unflatten directly from the fossilized representation rather than
+// deserializing everything first. It is the fastest option
+#define DIRECT_FROM_FOSSIL 0
+
 FIDDLE()
 namespace Slang
 {
 
+// The oldest compilers we support don't have std::bit_cast, so implement it
+// here
+template<typename To, typename From>
+    requires(sizeof(To) == sizeof(From)) && std::is_trivially_copyable_v<From> &&
+            std::is_trivially_copyable_v<To> && std::is_trivially_constructible_v<To>
+To bitCast(const From& src)
+{
+    To dst;
+    memcpy(&dst, &src, sizeof(To));
+    return dst;
+}
+
 //
 // We wrap everything up in an IRModuleInfo, to prepare for the case in which
 // we want to serialize some sidecar information to help with on-demand loading
 // or backwards compat
 //
+// For all the aggregate structs here we'll use Fiddle to generate the
+// deserialization code.
+//
 FIDDLE()
 struct IRModuleInfo
 {
     FIDDLE(...)
-    // Include the specific compiler version in serialized output, in case we
-    // ever need to do any version specific workarounds.
-    FIDDLE() String fullVersion = SLANG_TAG_VERSION;
     // Include this here so that if we need to change the way we serialize
     // things and maintain backwards compat we can increment this value, for
     // example if we introduce more instructions with weird payloads like
     // IRModuleInst or IRConstants.
-    const static UInt kSupportedSerializationVersion = 0;
-    FIDDLE() UInt serializationVersion = kSupportedSerializationVersion;
+    // If we want to support back compat we'll need to change this to a list of
+    // accepted values, and branch on that later down.
+    const static UInt64 kSupportedSerializationVersion = 999000000000999;
```

Comment:
is there a special meaning for this number?

Reply:
ah, that was a shibboleth for debugging, it's changed now to 1, good catch